pkg/controller/backupscheduled/backupscheduled_controller.go

```diff
@@ -0,0 +1,175 @@
+// +kubebuilder:rbac:groups=extension.shepherd,resources=backupscheduleds/finalizers,verbs=get;list;watch;create;update;patch;delete
+func (r *ReconcileBackupScheduled) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log := logger.New(ControllerName, request.Namespace, request.Name)
+	log.Info("Starting reconcile loop")
+
+	backupScheduled := &extensionv1.BackupScheduled{}
+	err := r.Get(context.TODO(), request.NamespacedName, backupScheduled)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	if backupScheduled.Spec.Schedule == "" {
+		err := errors.New("BackupScheduled doesn't have a schedule.")
+		log.Error(err.Error())
+		return reconcile.Result{}, err
+	}
+
+	if _, found := backupScheduled.ObjectMeta.GetLabels()["site"]; !found {
+		err := errors.New("BackupScheduled doesn't have a site label.")
+		log.Error(err.Error())
+		return reconcile.Result{}, err
+	}
+
+	log.Info("Calculating next scheduled backup.")
+	now := time.Now()
+	status := extensionv1.BackupScheduledStatus{
+		LastExecutedTime: &metav1.Time{Time: getScheduleComparison(backupScheduled.Status, now)},
+	}
+
+	// Check if we are currently due for a backup to be scheduled.
+	expr, err := cronexpr.Parse(backupScheduled.Spec.Schedule)
+	if err != nil {
+		log.Errorf("Cron schedule syntax is invalid: '%s'.", err.Error())
+		return reconcile.Result{}, err
+	}
+	next := expr.Next(status.LastExecutedTime.Time)
+	if next.Before(now) || next.Equal(now) {
+		// Due for a backup - create object.
+		log.Info("Backup due - creating.")
+		backup := &extensionv1.Backup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%d", backupScheduled.Name, now.Unix()),
+				Namespace: backupScheduled.Namespace,
+				Labels:    backupScheduled.Labels,
+				Annotations: map[string]string{
+					restic.FriendlyNameAnnotation: now.Format("Mon, 02/01/2006 - 15:04"),
+				},
+			},
+			Spec: extensionv1.BackupSpec{
+				MySQL:   backupScheduled.Spec.MySQL,
+				Volumes: backupScheduled.Spec.Volumes,
+			},
+		}
+
+		result, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, backup, sync.Backup(backupScheduled, r.scheme))
+		if err != nil {
+			return reconcile.Result{
+				Requeue: true,
+			}, err
+		}
+		log.Info(fmt.Sprintf("Created backup with result: %s.", result))
+
+		// Update status with LastExecutedTime.
+		status.LastExecutedTime.Time = time.Now()
+	}
+
+	if diff := deep.Equal(backupScheduled.Status, status); diff != nil {
+		log.Info(fmt.Sprintf("Status change dectected: %s", diff))
+
+		backupScheduled.Status = status
+
+		err := r.Status().Update(context.TODO(), backupScheduled)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to update status")
+		}
+	}
+
+	next = cronexpr.MustParse(backupScheduled.Spec.Schedule).Next(now)
+	nextDuration := next.Unix() - now.Unix()
```

Comment:
Wonder if this should go on the status! That would be a nice little field to expose in a web ui.